### PR TITLE
HCF-1253 trigger rebuild on patch change

### PR DESCRIPTION
**Pull request by alexjh**

### CODE REVIEW

File: model/roles.go [hunk 1]
```diff
@@ -168,8 +174,57 @@ func (r *Role) GetScriptPaths() map[string]string {
...
 }
 
+// GetScriptSignatures returns the SHA1 of all of the script file names and contents
+func (r *Role) GetScriptSignatures() (string, error) {
+	hasher := sha1.New()
+
+	var scripts []string
+
+	for _, f := range r.GetScriptPaths() {
+		scripts = append(scripts, f)
```
Comment (aarondl) [inline model/roles.go]: Use copy built in here instead of a for loop + append.

File: model/roles_test.go [hunk 2]
```diff
@@ -197,6 +198,88 @@ func TestRolesSort(t *testing.T) {
...
+	firstHash, _ := refRole.GetScriptSignatures()
+
+	workDir, err := ioutil.TempDir("", "fissile-test-")
+	assert.NoError(err)
+	defer os.RemoveAll(workDir)
+	releasePath := filepath.Join(workDir, "role.yml")
+
+	scriptName := "script.sh"
+	scriptPath := filepath.Join(workDir, scriptName)
+	scriptHandle, err := os.Create(scriptPath)
```
Comment (aarondl) [inline model/roles_test.go]: Probably want to use [ioutil.WriteFile](https://golang.org/pkg/io/ioutil/#WriteFile)

File: model/roles_test.go [hunk 3]
```diff
@@ -197,6 +198,88 @@ func TestRolesSort(t *testing.T) {
...
+		Scripts: []string{scriptName},
+		rolesManifest: &RoleManifest{
+			manifestFilePath: releasePath,
+		},
+	}
+
+	differentPatchHash, _ := differentPatch.GetScriptSignatures()
+	assert.NotEqual(firstHash, differentPatchHash, "role hash should be dependent on patch string")
+
+	scriptHandle, err = os.Create(scriptPath)
```
Comment (aarondl) [inline model/roles_test.go]: Same as above

File: model/roles.go (hunk 1 above)
Comment (alexjh) [inline model/roles.go]: `GetScriptPaths()` returns a map, so using a loop was the only way. Made it a bit more efficient by allocating the array ahead of time.

File: model/roles_test.go (hunk 2 above)
Comment (alexjh) [inline model/roles_test.go]: Updated in both places.

**Comment by viovanov** [comment]
@alexjh can you rebase this please?

**Comment by alexjh** [comment]
Rebased on master, CI checks are passing.

**Pull request merged**